gateway/ln-gateway/src/lib.rs

```diff
@@ -901,6 +913,17 @@ impl Gateway {
             self.client_builder
                 .save_config(gw_client_cfg.clone(), dbtx)
                 .await?;
+            let scids: Vec<u64> = self
+                .scid_to_federation
+                .read()
+                .await
+                .keys()
+                .cloned()
+                .collect();
+
+            if let Err(e) = lnrpc.update_scids(scids).await {
+                warn!("Failed to update scids: {:?}", e);
+            }
```

Comment:
Code duplication, can this become a function?

Reply:
moved into function